drivers/filesystems/fastfat/fcb.c

```diff
@@ -310,10 +310,24 @@ vfatReleaseFCB(
 
     while (pFCB)
     {
+        ULONG RefCount;
+
         ASSERT(pFCB != pVCB->VolumeFcb);
         ASSERT(pFCB->RefCount > 0);
-        pFCB->RefCount--;
-        if (pFCB->RefCount == 0)
+        RefCount = --pFCB->RefCount;
+
+        if (RefCount == 1 && BooleanFlagOn(pFCB->Flags, FCB_CACHE_INITIALIZED))
```

Comment:
Such CACHE_INITIALIZED checks always look a little weird to me since you're supposed to call CcUninitializeCacheMap unconditionally, i.e. even if CcInitializeCacheMap wasn't called on that file.

Reply:
This is not exactly the same here, it's for the stream file object that is allocated for every directories.
It avoids attempting to uninitialize whereas the directory already went through stream file object deletion.
It's safe, and avoids  other (and less consistent) checks. See initial patches attached to the bug report to see alternatives.

Reply:
Yeah it certainly makes sense to avoid doing the ObDereferenceObject when you're not supposed to. I guess I was just remarking that the flag in general is a bit odd.

Reply:
Well, other attempts without that flag lead to double frees and such. So, it's rather a good workaround (yeah, it's not perfect).